src/ducks/apps/index.js

```diff
@@ -195,78 +195,12 @@ function _sortAlphabetically(array, property) {
 /* Only for the icon fetching */
 const root = document.querySelector('[role=application]')
 const data = root && root.dataset
-const COZY_TOKEN = data && data.cozyToken
 const COZY_DOMAIN = data && `//${data.cozyDomain}`
 /* Only for the icon fetching */
 
-const mimeTypes = {
-  gif: 'image/gif',
-  ico: 'image/vnd.microsoft.icon',
-  jpeg: 'image/jpeg',
-  jpg: 'image/jpeg',
-  png: 'image/png',
-  svg: 'image/svg+xml'
-}
-
-export const fetchIcon = (app = {}) => async url => {
-  let icon
-  try {
-    const resp = await fetch(`${COZY_DOMAIN}${url}`, {
-      method: 'GET',
-      credentials: 'include',
-      headers: {
-        Authorization: `Bearer ${COZY_TOKEN}`
-      }
-    })
-    if (!resp.ok)
-      throw new Error(`Error while fetching icon ${resp.statusText}: ${url}`)
-    icon = await resp.blob()
-  } catch (error) {
-    throw error
-  }
-  if (!icon.type) {
-    // iOS10 does not set correctly mime type for images, so we assume
-    // that an empty mime type could mean that the app is running on iOS10.
-    // For regular images like jpeg, png or gif it still works well in the
-    // Safari browser but not for SVG.
-    // So let's set a mime type manually. We cannot always set it to
-    // image/svg+xml and must guess the mime type based on the icon attribute
-    // from app/manifest
-    // See https://stackoverflow.com/questions/38318411/uiwebview-on-ios-10-beta-not-loading-any-svg-images
-
-    const manifestIcon =
-      (app.latest_version &&
-        app.latest_version.manifest &&
-        app.latest_version.manifest.icon) ||
-      app.icon
-
-    if (!manifestIcon) {
-      throw new Error(`${app.name}: Cannot detect mime type for icon ${url}`)
-    }
-
-    const extension = manifestIcon.split('.').pop()
-
-    if (!extension) {
-      throw new Error(
-        `${
-          app.name
-        }: Unable to detect icon mime type from extension (${manifestIcon})`
-      )
-    }
-
-    if (!mimeTypes[extension]) {
-      throw new Error(
-        `${app.name}: 'Unexpected icon extension (${manifestIcon})`
-      )
-    }
-
-    icon = new Blob([icon], { type: mimeTypes[extension] })
-  }
-
-  if (icon.type.match(/^image\/.*$/)) {
-    return URL.createObjectURL(icon)
-  }
-  throw new Error(`${app.name}: icon ${url} is not an image.`)
+export const getIconUrl = app => {
+  if (!app || !app.links) return null
+  return `${COZY_DOMAIN}${app.links.icon}`
```

Comment:
This will not work with registry apps

Reply:
As we get the `COZY_DOMAIN` here we are really close to pass it directly to `AppIcon` as prop. `AppIcon` then handles correctly either installed apps and registry apps.

Reply:
Actually this works with the registry since the Store make the two sources 'iso' with https://github.com/cozy/cozy-store/blob/master/src/ducks/apps/index.js#L562
But I agree, we can make things simpler, I changed it to use directly domain and the secure props 👍